lib/teleterm/services/unifiedresources/unifiedresources.go

```diff
@@ -92,26 +93,14 @@ func List(ctx context.Context, cluster *clusters.Cluster, client Client, req *pr
 					App:      app,
 				},
 			})
-		case *proto.PaginatedResource_AppServerOrSAMLIdPServiceProvider:
-			if e.AppServerOrSAMLIdPServiceProvider.IsAppServer() {
-				app := e.AppServerOrSAMLIdPServiceProvider.GetAppServer().GetApp()
-				response.Resources = append(response.Resources, UnifiedResource{
-					App: &clusters.App{
-						URI:      cluster.URI.AppendApp(app.GetName()),
-						FQDN:     cluster.AssembleAppFQDN(app),
-						AWSRoles: cluster.GetAWSRoles(app),
-						App:      app,
-					},
-				})
-			} else {
-				provider := e.AppServerOrSAMLIdPServiceProvider.GetSAMLIdPServiceProvider()
-				response.Resources = append(response.Resources, UnifiedResource{
-					SAMLIdPServiceProvider: &clusters.SAMLIdPServiceProvider{
-						URI:      cluster.URI.AppendApp(provider.GetName()),
-						Provider: provider,
-					},
-				})
-			}
+		case *proto.PaginatedResource_SAMLIdPServiceProvider:
+			provider := e.SAMLIdPServiceProvider
+			response.Resources = append(response.Resources, UnifiedResource{
+				SAMLIdPServiceProvider: &clusters.SAMLIdPServiceProvider{
+					URI:      cluster.URI.AppendApp(provider.GetName()),
+					Provider: provider,
+				},
+			})
```

Comment:
This is a breaking change. If my auth server is on v15.0.0 and then I update Connect to, let's say v15.2, I won't see SAML apps. I believe we should support both `proto.PaginatedResource_AppServerOrSAMLIdPServiceProvider` and `proto.PaginatedResource_SAMLIdPServiceProvider` here if we want to backport it.

Reply:
This is legit. Thanks for flagging, completely missed the fact the Connect compatibility is entirely different than the Web UI.

Though, if I'll have to keep both the types, then there is a little benefit to have this backported to `v15` but let me think about this one.

Reply:
I spent some time thinking and trying out alternative for this one. Even when we support both in this PR, once we backport to `v15`, there will be another breaking change if Auth and Connect are not updated at to the latest version the same time. 

Because when Auth is on higher version than Connect, Auth will start sending `types.SAMLIdPServiceProvider` instead of `types.AppServerOrSAMLIdPServiceProvider` and the lower versioned Connect will not recognize the `SAMLIdPServiceProvider` type. There wont be an error but users will not see SAML apps.

This may have ben averted if there was an additional response layer set up in auth, that would process the type and send SAML app as "app", similar to how we do with Web UI.

Now since it is always better to do breaking change in a major version release, i.e. the `16.0` in this case, I am leaning towards not backporting this PR. Though, thinking it other way, we are still a bit early in the `v15` so I'll probably tag you, @avatus @rudream here. What do you all think? 

Let me know if I miss something @gzdunek

Reply:
>Because when Auth is on higher version than Connect, Auth will start sending types.SAMLIdPServiceProvider instead of types.AppServerOrSAMLIdPServiceProvider and the lower versioned Connect will not recognize the SAMLIdPServiceProvider type.

Oh, right. 
It looks like the only way to maintain the backward compatibility would be sending both types from the server and also handling both of them on the client side; but we would end up with duplicate resources.

I'm fine with merging this PR only to v16, but to be precise, [v16 servers should still support v15 clients](https://goteleport.com/docs/upgrading/overview/#component-compatibility). I don't have a good answer what to do with it, OTOH SAML apps are not that useful in Connect, the only thing you can do with them is to open them in the browser where you have to log in anyway (by that I want to say that maybe we can live with that v16/v15 incompatibility).

Reply:
So the deprecated types handling are added back and this PR will not be backported to `v15`. https://github.com/gravitational/teleport/pull/38709/commits/3b0579af7e82f033cd4d50fa4c5bd9382445e594

To prepare Connect for breaking changes, the plan is: to selectively backport the teleterm package to `v15`. So the Connect client in v15 can process both the `types.AppServerOrSAMLIdPServiceProvide` and `types.SAMLIdPServiceProvider` types.

Once the changes for Auth Service are released in v16, connect should be ready to process the type changes.